# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/audioipc/audioipc/src/messagestream_win.rs
# Commit: 0a7873a6b522
# Full Hash: 0a7873a6b522ce4aef9a32ae4a47f093ba33e0f4
# Author: Matthew Gregan <kinetik@flim.org>
# Date: 2020-01-26 09:48:16
# Regressor Bug: 1590249
# File Overlap Count: 3
# Description:
#   Bug 1590249 - Update audioipc to 86d49ddf.  r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D60615
# ==============================================================================

diff -r 36f6505b50c9 -r 0a7873a6b522 media/audioipc/audioipc/src/messagestream_win.rs
--- a/media/audioipc/audioipc/src/messagestream_win.rs	Sat Jan 25 21:30:14 2020 +0000
+++ b/media/audioipc/audioipc/src/messagestream_win.rs	Sat Jan 25 21:31:32 2020 +0000
@@ -8,7 +8,7 @@
 use std::os::windows::io::{AsRawHandle, FromRawHandle, IntoRawHandle, RawHandle};
 use std::sync::atomic::{AtomicUsize, Ordering};
 use tokio_io::{AsyncRead, AsyncWrite};
-use tokio_named_pipes;
+use super::tokio_named_pipes;
 use winapi::um::winbase::FILE_FLAG_OVERLAPPED;
 
 #[derive(Debug)]
@@ -23,8 +23,8 @@
     pub fn anonymous_ipc_pair(
     ) -> std::result::Result<(MessageStream, MessageStream), std::io::Error> {
         let pipe_name = get_pipe_name();
-        let pipe1 = miow::pipe::NamedPipe::new(&pipe_name)?;
-        let pipe2 = {
+        let pipe_server = miow::pipe::NamedPipe::new(&pipe_name)?;
+        let pipe_client = {
             let mut opts = std::fs::OpenOptions::new();
             opts.read(true)
                 .write(true)
@@ -32,7 +32,7 @@
             let file = opts.open(&pipe_name)?;
             unsafe { miow::pipe::NamedPipe::from_raw_handle(file.into_raw_handle()) }
         };
-        Ok((MessageStream::new(pipe1), MessageStream::new(pipe2)))
+        Ok((MessageStream::new(pipe_server), MessageStream::new(pipe_client)))
     }
 
     pub unsafe fn from_raw_fd(raw: super::PlatformHandleType) -> MessageStream {
@@ -50,6 +50,12 @@
     }
 }
 
+impl IntoRawHandle for MessageStream {
+    fn into_raw_handle(self) -> RawHandle {
+        self.0.into_raw_handle()
+    }
+}
+
 impl AsyncMessageStream {
     fn new(stream: tokio_named_pipes::NamedPipe) -> AsyncMessageStream {
         AsyncMessageStream(stream)
@@ -93,12 +99,6 @@
     }
 }
 
-impl IntoRawHandle for MessageStream {
-    fn into_raw_handle(self) -> RawHandle {
-        self.0.into_raw_handle()
-    }
-}
-
 static PIPE_ID: AtomicUsize = AtomicUsize::new(0);
 
 fn get_pipe_name() -> String {